zellij-tile/src/lib.rs

```diff
@@ -1,13 +1,54 @@
 pub mod prelude;
 pub mod shim;
 
+use zellij_utils::anyhow::{self, Result};
 use zellij_utils::data::Event;
 
 #[allow(unused_variables)]
 pub trait ZellijPlugin {
-    fn load(&mut self) {}
-    fn update(&mut self, event: Event) {}
-    fn render(&mut self, rows: usize, cols: usize) {}
+    fn load(&mut self) -> Result<()> {
+        Ok(())
+    }
+    fn update(&mut self, event: Event) -> Result<()> {
+        Ok(())
+    }
+    fn render(&mut self, rows: usize, cols: usize) -> Result<()> {
+        Ok(())
+    }
+}
+
+/// Catch non-fatal errors and print their content to the plugins stdout and stderr. This way, the
+/// user sees the error and can even report it, but it will not crash the application. Hence, this
+/// works only on the empty type `()`.
+pub fn catch(ret: anyhow::Result<()>) {
+    match ret {
+        Ok(_) => (),
+        Err(err) => {
+            println!("ERROR: {}", err);
+            eprintln!("ERROR: {}", &err);
+            err.chain().skip(1).for_each(|cause| {
+                let msg = format!("because: {cause}");
+                println!("{}", msg);
+                eprintln!("{}", &msg);
+            });
+        },
+    }
+}
+
+/// Return the inner value if `Ok(_)` and panic the application otherwise. Before panicking, the
+/// error is written to the logfile.
+pub fn catch_fatal<T>(ret: anyhow::Result<T>) -> T {
+    match ret {
+        Ok(val) => val,
+        Err(err) => {
+            eprintln!("ERROR: {}", &err);
+            err.chain().skip(1).for_each(|cause| {
+                let msg = format!("because: {cause}");
+                eprintln!("{}", &msg);
+            });
+            panic!("{:#?}", err);
+        },
+    }
```

Comment:
@tlinford Maybe you have an idea how this bit of code can be written prettier? I wanted to put this into a trait and implement that on `std::result::Result` but I failed miserably. Makes the code below pretty unreadable, too...